AGENT.md

```diff
@@ -0,0 +1,112 @@
+
+### Individual Package Scripts
+
+Each package in `packages/` has these common scripts:
+- `yarn build` - Build the specific package
+- `yarn clean` - Clean build artifacts  
+- `yarn dev` - Watch mode for development
+- `yarn lint` - Check linting rules only
+- `yarn lint:fix` - Fix linting issues automatically
+- `yarn typecheck` - Type check without emitting files
+
+## Architecture
+
+### Monorepo Structure
+
+The project follows a monorepo pattern with these core packages:
+
+1. **`@vertex-protocol/client`** - Main entry point that composes all other packages into a unified `VertexClient`
+2. **`@vertex-protocol/engine-client`** - Handles off-chain matching engine communication
+3. **`@vertex-protocol/indexer-client`** - Provides indexer queries for historical data
+4. **`@vertex-protocol/trigger-client`** - Manages trigger service for stop orders
+5. **`@vertex-protocol/contracts`** - Contract utilities, ABIs, and on-chain interactions
+6. **`@vertex-protocol/utils`** - Common utilities including BigNumber.js for decimal math
+
+### Client Architecture
+
+- `VertexClient` is the main class that orchestrates all API interactions
+- Uses `viem` for Ethereum wallet/provider functionality (migrated from ethers)
+- Supports both chain signers and linked signers for trading
+- Modular API design with separate classes for Market, Spot, Perp, Subaccount, and WebSocket operations
+
+### Key Patterns
+
+- All packages use TypeScript with strict type checking
+- BigDecimal/BigNumber.js for precise decimal calculations
+- EIP-712 signing for off-chain order execution
+- Comprehensive type definitions for all API responses
+- Consistent error handling with custom error classes
+- Viem as the primary Ethereum library dependency
+
+## Test and Verification Sequence
+
+After making edits, **ALWAYS** run the following verification sequence:
+
+1. **Type Check**  
+   - Run `yarn typecheck` to verify all TypeScript types are correct across all packages.
+2. **Lint Check**  
+   - Run `yarn lint` to run ESLint with auto-fix and Prettier formatting.
+3. **Build**  
+   - Run `yarn build` to build all packages before running any tests.
+4. **Tests**
+   - Run `yarn test` to run all tests across the codebase.
+5. **E2E Tests**
+   - Run **E2E tests** to confirm core flows are working.
+   - Add new **basic sanity E2E tests** for any new functionality introduced.
+
+### Requirements
+- **All commands must pass** before considering a task complete
+- **Fix errors immediately** - If any command fails, address issues and re-run the full sequence
+- **Build before E2E** - Always run `yarn build` before E2E testing to ensure packages are properly built
+- **Add basic sanity E2E tests** for new features and when modifying client APIs, core functionality, or user flows
+- **No unit tests** - This project uses E2E integration tests in `apps/e2e`, any unit tests should be written manually
```

Comment:
Could this be mistaken for the fact that we don't have any unit tests? (We do, we just need to expand on them)